src/jit/lowerxarch.cpp

```diff
@@ -3763,6 +3867,63 @@ bool Lowering:: IsContainableImmed(GenTree* parentNode, GenTree* childNode)
     return true;
 }
 
+//----------------------------------------------------------------------
+// SetRegOptional - sets a bit to indicate to LSRA that register for
+// for a given tree node is optional for codegen purpose.
+//
+// Arguments:
+//    tree    -   GenTree node
+//
+// Returns
+//    None
+//
+// Note: Right now a tree node is marked as reg optional only
+// if is it a GT_LCL_VAR.  This routine needs to be modified if
+// in future if lower/codegen needs to support other tree node
+// types.
+void Lowering::SetRegOptional(GenTree* tree)
```

Comment:
Here's where it would be useful to talk about the fact that, if it is marked as reg optional, it will be treated as contained in code generation, i.e. it will be used as a memory operand by its parent.